### PR TITLE
Vuldet passes severity and cvss scores to insert helper

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#8773|
## Description
This PR implements the changes needed for the vulnerability detector to be able to send the severity, and cvss scores to wazuh-db so it can be inserted in the vuln_cves inventory.

## Dod
**Agents**

Centos 8 (Manager)

Ubuntu Bionic

Windows Server 2019

## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
- [X] Source installation
- [X] Source upgrade
- [X] Review logs syntax and correct language

### CODE REVIEW

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -1361,34 +1360,7 @@ int wm_vuldet_process_agent_vulnerabilities(sqlite3 *db, OSHash *cve_table, scan
...
-                        scan_ctx->agent_id);
-            }
-
-            if (scan_ctx->scan_type == VU_BASELINE_SCAN || update) {
-                // No report should be generated for baseline scans or already reported vulnerabilities
-                pkg = next;
-                continue;
-            }
-
+            wm_vuldet_free_report(report);
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1363]: If you move the call of `wm_vuldet_free_report` to this place, you will cause always a memory leak of the last report being evaluated in the `do while` loop.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -1436,6 +1408,36 @@ int wm_vuldet_process_agent_vulnerabilities(sqlite3 *db, OSHash *cve_table, scan
                 }
             }
 
+            //Save the vulnerability in the agent database
+            bool check_pkg_existence = !strcmp(pkg->type, VULN_CVES_TYPE_PACKAGE);
+            cJSON* j_result = wdb_insert_vuln_cves(scan_ctx->agent_id, pkg->bin_name, pkg->version, pkg->arch, cve,
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1413]: This is not related to your implementation, but I discovered that we have a potential issue with the string of the version that we are using. If you take a look at the piece of code that fills the `report->version` field, it does something like this...

```c
            if (!strncmp("-1:", pkg->version, 3)) {
                w_strdup((pkg->version + 3), report->version);
            } else {
                w_strdup(pkg->version, report->version);
            }
```

So if we make use of `pkg->version`, we will not be handling the case of versions starting with `-1:`. The result will be a difference between the versions in the alerts and the inventory.

My suggestion here is to change  `pkg->bin_name` with `report->software`, also `pkg->version` with `report->version`, and `pkg->arch` with `report->arch`. So in this way we make sure that the inventory will have exactly the same information reported in the alerts.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, apply changes according to the comments.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 1 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: Totally! Fixed now

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 2 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: Indeed is related to these changes, I did move that block of code. Good catch, thanks

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**